File: app/src/main/java/com/youtube/sorcjc/redemnorte/ui/fragment/MessageDialogFragment.java
package com.youtube.sorcjc.redemnorte.ui.fragment;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

public class MessageDialogFragment extends DialogFragment {
    public interface MessageDialogListener {
        void onDialogPositiveClick(DialogFragment dialog);
        void onDialogNeutralClick(DialogFragment dialog);
        void onDialogNegativeClick(DialogFragment dialog);
    }

    private String mTitle;
    private String mMessage;
    private MessageDialogListener mListener;

    public void onCreate(Bundle state) {
        super.onCreate(state);
        setRetainInstance(true);
    }

    public static MessageDialogFragment newInstance(String title, String message, MessageDialogListener listener) {
        MessageDialogFragment fragment = new MessageDialogFragment();
        fragment.mTitle = title;
        fragment.mMessage = message;
        fragment.mListener = listener;
        return fragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setMessage(mMessage)
                .setTitle(mTitle);

        builder.setPositiveButton("Usar código", new DialogInterface.OnClickListener() {
            public void onClick(DialogInterface dialog, int id) {
                if (mListener != null) {
                    mListener.onDialogPositiveClick(MessageDialogFragment.this);
                }
            }
        });
        builder.setNeutralButton("Volver a tomar", new DialogInterface.OnClickListener() {
            public void onClick(DialogInterface dialog, int id) {
                if (mListener != null) {
                    mListener.onDialogNeutralClick(MessageDialogFragment.this);
                }
            }
        });
        builder.setNegativeButton("Cancelar", new DialogInterface.OnClickListener() {
            public void onClick(DialogInterface dialog, int id) {
                if (mListener != null) {
                    mListener.onDialogNegativeClick(MessageDialogFragment.this);
                }
            }
        });

        return builder.create();
    }
}

File: app/src/main/java/com/youtube/sorcjc/redemnorte/ui/fragment/MessageDialogFragment.kt
package com.youtube.sorcjc.redemnorte.ui.fragment

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.youtube.sorcjc.redemnorte.R
import com.youtube.sorcjc.redemnorte.ui.fragment.MessageDialogFragment

class MessageDialogFragment : DialogFragment() {
    interface MessageDialogListener {
        fun onDialogPositiveClick(dialog: DialogFragment?)
        fun onDialogNeutralClick(dialog: DialogFragment?)
        fun onDialogNegativeClick(dialog: DialogFragment?)
    }

    private var mTitle: String? = null
    private var mMessage: String? = null
    private var mListener: MessageDialogListener? = null
    override fun onCreate(state: Bundle?) {
        super.onCreate(state)
        retainInstance = true
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity!!)
        builder.setMessage(mMessage)
                .setTitle(mTitle)
        builder.setPositiveButton(getString(R.string.btn_positive_use_code)) { dialog, id ->
            if (mListener != null) {
                mListener!!.onDialogPositiveClick(this@MessageDialogFragment)
            }
        }
        builder.setNeutralButton(getString(R.string.btn_neutral_take_again)) { dialog, id ->
            if (mListener != null) {
                mListener!!.onDialogNeutralClick(this@MessageDialogFragment)
            }
        }
        builder.setNegativeButton(getString(R.string.btn_negative_cancel)) { dialog, id ->
            if (mListener != null) {
                mListener!!.onDialogNegativeClick(this@MessageDialogFragment)
            }
        }
        return builder.create()
    }

    companion object {
        fun newInstance(title: String?, message: String?, listener: MessageDialogListener?): MessageDialogFragment {
            val fragment = MessageDialogFragment()
            fragment.mTitle = title
            fragment.mMessage = message
            fragment.mListener = listener
            return fragment
        }
    }
}